contracts/rewards/RewardsController.sol

```diff
@@ -227,24 +227,30 @@ contract RewardsController is RewardsDistributor, VersionedInitializable, IRewar
     if (amount == 0) {
       return 0;
     }
-    uint256 unclaimedRewards = _usersUnclaimedRewards[user][reward];
+    uint256 totalRewards;
 
-    if (amount > unclaimedRewards) {
-      _distributeRewards(user, _getUserStake(assets, user));
-      unclaimedRewards = _usersUnclaimedRewards[user][reward];
+    _distributeRewards(user, _getUserStake(assets, user));
+    for (uint256 i = 0; i < assets.length; i++) {
```

Comment:
There is an issue here. The `totalRewards` var is never reduced, so the difference may be larger than the reward accrued for a specific asset. This can lead to underflow and thereby revert. See the example below:

Example: 
```python
rewardsAccrued = [1000, 100, 50, 1000, 150]
totalRewards = 0
amount = 1100

for i in range(len(rewardsAccrued)):
    totalRewards = totalRewards + rewardsAccrued[i]
    
    if totalRewards <= amount:
        rewardsAccrued[i] = 0
    else:
        difference = totalRewards - amount
        rewardsAccrued[i] = rewardsAccrued[i] - difference

Values after:
TotalRewards  :  2300
rewardsAccrued:  [0, 0, 0, -50, -1050]
```

Reply:
mmh, there is a break so the loop ends the first time totalRewards > amount, therefore the -1050 should never happen

But you are right that totalRewards should be decreased by difference to match the correct amount to transfer. And also in the else, `rewardsAccrued[i]` should be equal to `difference`, not to `rewardsAccrued[i] - difference` as the remainings are the surplus against rewardsAccrued

Example

```
rewardsAccrued = [1000, 80, 50, 1000, 150]
totalRewards = 0
amount = 1100

for i in range(len(rewardsAccrued)):
    totalRewards = totalRewards + rewardsAccrued[i]
    
    if totalRewards <= amount:
        rewardsAccrued[i] = 0
    else:
        difference = totalRewards - amount
        rewardsAccrued[i] = difference
        break

Values after:
TotalRewards  :  1100
rewardsAccrued:  [0, 0, 30, 100, 1000]

Reply:
Oh. Missed the break. 
Think it is good with the change to difference. But you example is wrong on the `totalRewards` value and `rewardsAccrued` list in the end. It should be:
```
totalRewards  :  1130
rewardsAccrued:  [0, 0, 30, 1000, 150]
```